src/js/components/Diagram/Diagram.js

```diff
@@ -145,16 +159,52 @@ const Diagram = forwardRef(({ connections, ...rest }, ref) => {
             toPoint[0] += toRect.width / 2;
             if (fromRect.top < toRect.top) {
               fromPoint[1] += fromRect.height;
+
+              if (typeof endpoint === 'object' && endpoint?.from) {
+                fromPoint[1] += 10;
```

Comment:
How was `10` determined? It might be helpful to add a code comment for future.

Reply:
Sure, will do